flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/OperationTreeBuilder.scala

```diff
@@ -376,6 +377,33 @@ class OperationTreeBuilder(private val tableEnv: TableEnvironment) {
     Project(expandedFields.map(UnresolvedAlias), childNode).validate(tableEnv)
   }
 
+  def flatMap(tableFunction: Expression, child: TableOperation): TableOperation = {
```

Comment:
Maybe this method can be simplified as follows ?
```
  def flatMap(tableFunction: Expression, child: TableOperation): TableOperation = {
    val tableFuncWithoutAlias = getLeadingNonAliasExpr(tableFunction)
    val joinNode = joinLateral(child, tableFuncWithoutAlias, JoinType.INNER, Optional.empty())
    val dropNode = dropColumns(
      child.getTableSchema.getFieldNames.map(a => UnresolvedFieldReference(a)),
      joinNode)

    val temporalTable = UserDefinedFunctionUtils.createLogicalFunctionCall(
      expressionBridge.bridge(tableFunction),
      child.asInstanceOf[LogicalNode])

    alias(temporalTable.fieldNames.map(a => UnresolvedFieldReference(a)), dropNode)
  }
```
Furthermore, we don't need the `extractTableFunction()` in `UserDefinedFunctionUtils` if we change the implementation to the above one. 
What do you think?

Reply:
Good advice for using dropNode to remove the columns from the original inputs. Regarding to the logic to solve duplicate field names, I think it is still needed as input field names may be duplicate with the result field names of table function. Thoughts?